spec/design/skel/ARM_HYP/ArchInterruptDecls_H.thy

```diff
@@ -8,9 +8,12 @@ theory ArchInterruptDecls_H
 imports RetypeDecls_H CNode_H
 begin
 
+requalify_types (in Arch)
+  sgisignal_invocation
+
 context Arch begin arch_global_naming (H)
 
-#INCLUDE_HASKELL SEL4/Object/Interrupt/ARM.lhs CONTEXT Arch decls_only ArchInv=
+#INCLUDE_HASKELL SEL4/Object/Interrupt/ARM.lhs CONTEXT Arch decls_only ArchInv=Arch
```

Comment:
do you remember what this does again, and why you needed to add it?

Reply:
Ok, I looked it up: Haskell sometimes imports arch definitions `as Arch` and sometimes `as ArchInv` (presumably when they are invocations). The line here tells the Haskell translator how to translate `ArchInv.` when it sees it.

Which means, the declaration is not necessary in this file, because this is an arch specific file and doesn't need to refer to `ArchInv.`.  Most likely a left-over from importing one of the functions in the wrong place before arch splitting. Will remove.